### PR TITLE
version 3.0.0

### PR DESCRIPTION
**Pull request by Equbuxu**
Separated the color picker into 4 controls:

- Color display
- Hex textbox
- HSV/HSL color picker
- RGBA/HSV/HSL sliders

other changes:

- Added HSL color space, made it togglable on HSV/HSL color picker aka SquarePicker
- Recreated StandardColorPicker and PortableColorPicker using those controls
- Redesigned controls code-behind to allow easy binding and creation of compatible third-party controls
- Added a demo project
- Rewritten the entire readme
- Added ShowAlpha property that toggles alpha channel visibility
- Removed System.Windows.Interactivity dependency
- Added support for .net framework 4.5+ and .net core 3.1
- Fixed gradient alignment on sliders
- Added package icon and updated its description
- Made slider handles ellipses rather than images
- Made all controls scalable
- Add ColorChanged event to be used instead of bindings
- Added a function to swap colors

good luck reviewing those 46 commits

### CODE REVIEW

**Review by CPKreuz — CHANGES_REQUESTED** [review]
When entering something in the text box of the alpha slider and then hitting enter doesn't unfocus the text box

When you disable the alpha slider and enter something in the text box above the non existing alpha slider you can hit enter to enter something in the alpha text box

**Comment by Equbuxu** [comment]
Alpha slider and textbox now get disabled when hidden. I made the textboxes lose focus completely instead of switching to the next element, it makes more sense to me as you would probably use the hex field if your goal is to change all values

**Comment by CPKreuz** [comment]
> it makes more sense to me as you would probably use the hex field if your goal is to change all values

Ah yes, lemme just convert that decimal value to a hex value. No problem for me. 


But it does make more sense generally

File: README.md [hunk 1]
```diff
@@ -3,40 +3,79 @@
 
 # About
 
-A WPF Color picker made for [PixiEditor](https://github.com/PixiEditor/PixiEditor). Available to use for anyone, anywhere!
+A collection of WPF controls the let users choose colors in various ways.
```
Comment (CPKreuz) [inline README.md:6]: Did you mean `that let`

**Review by CPKreuz — CHANGES_REQUESTED** [review]

File: src/ColorPicker/Models/MathHelper.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+﻿namespace ColorPicker.Models
+{
+    internal static class MathHelper
+
+    {
+        public static double Clamp(double value, double min, double max)
```
Comment (flabbet) [inline src/ColorPicker/Models/MathHelper.cs:6]: Why not use `Math.Clamp`?

Comment (Equbuxu) [reply src/ColorPicker/Models/MathHelper.cs:6]: There is no Math.Clamp in .NET Framework sadly

File: README.md (hunk 1 above)
Comment (Equbuxu) [reply README.md]: Yes

**Comment by flabbet** [comment]
Shouldn't color picker switch to HSV and HSL automatically on HSL and HSV sliders change?

**Review by flabbet — COMMENTED** [review]
Code looks good

File: src/ColorPicker/ColorPicker.csproj [hunk 3]
```diff
@@ -1,42 +1,49 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net5.0-windows</TargetFramework>
+    <TargetFrameworks>net45;netcoreapp3.1</TargetFrameworks>
```
[the remark this answers is not in the document]
Comment (Equbuxu) [reply src/ColorPicker/ColorPicker.csproj]: yes, and now I'm targeting net5.0 explicitly

**Review by flabbet — APPROVED** [review]
Well done

**Review by CPKreuz — APPROVED** [review]
gj

**Pull request merged**